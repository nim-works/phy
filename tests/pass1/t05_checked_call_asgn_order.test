discard """
  description: "
    Make sure register allocation takes the evaluation in checked calls into
    account
  "
"""
(TypeDefs
  (Int 8)
  (ProcTy (Type 0) (Type 0))
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) 0
    (List
      (Block (Params (Type 0))
        (Locals)
        (Return (IntVal 100)))))

  (ProcDef (Type 2) 0
    (List
      (Block (Params)
        (Locals (Type 0) (Type 0))
        (Asgn (Local 0) (IntVal 100))
        (CheckedCallAsgn (Local 1) (Proc 0)
          (Move (Local 0))
          (Goto 1
            (List (Local 0)))
          (Unwind)))

      (Block (Params (Type 0))
        (Locals)
        (Return)))))
