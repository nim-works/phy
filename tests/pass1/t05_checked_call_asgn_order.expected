;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Type 0) (Type 0))
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) 0
    (Locals (Type 0))
    (List
      (Block (Params)
        (Return (IntVal 100)))))
  (ProcDef (Type 2) 0
    (Locals (Type 0))
    (List
      (Block (Params)
        (Asgn (Local 0) (IntVal 100))
        (CheckedCallAsgn (Local 0) (Proc 0)
          (Move (Local 0))
          (Goto 1)
          (Unwind)))
      (Block (Params) (Return)))))
